### PR TITLE
FluentD Grep filter support

### LINKED ISSUE #268: Ability to exclude logs
**Is your feature request related to a problem? Please describe.**

We receive a lot of logs with log level `debug` or `info`. While we're trying to reduce these to warn/error in the source system itself, it's not always possible, e.g. canal managed by kops.

**Describe the solution you'd like**

I would like a way to exclude logs based on certain criteria.

**Describe alternatives you've considered**

N/A aside from disabling it in the source system.

**Additional context**

This should be available via grep: https://docs.fluentd.org/filter/grep

### PR DESCRIPTION
**Pull request by ahma**
- Docs

| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| Related tickets | fixes #268 
| License         | Apache 2.0

- [x] Error handling code meets the [guideline](https://github.com/banzaicloud/pipeline/blob/master/docs/error-handling-guide.md)
- [x] Logging code meets the guideline (TODO)
- [x] User guide and development docs updated (if needed)
- [x] Related Helm chart(s) updated (if needed)

- [x] If the PR is not complete but you want to discuss the approach, list what remains to be done here
- [x] Test

### CODE REVIEW

File: pkg/sdk/model/filter/grep_test.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+import (
+	"testing"
+
+	"github.com/banzaicloud/logging-operator/pkg/sdk/model/filter"
+	"github.com/banzaicloud/logging-operator/pkg/sdk/model/render"
+	"github.com/ghodss/yaml"
+)
+
+func TestGrep(t *testing.T) {
```
Comment (pepov) [inline pkg/sdk/model/filter/grep_test.go]: please extend the test to cover exclude and the or directive as well

Comment (ahma) [reply pkg/sdk/model/filter/grep_test.go]: okay

Comment (ahma) [reply pkg/sdk/model/filter/grep_test.go]: done

File: pkg/sdk/model/filter/grep.go [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+// ```
+//apiVersion: logging.banzaicloud.io/v1beta1
+//kind: Flow
+//metadata:
+//  name: demo-flow
+//spec:
+//  filters:
+//    - grep:
+//        regexp:
+//        - key: elso
```
Comment (tarokkk) [inline pkg/sdk/model/filter/grep.go]:
```suggestion
//        - key: first
```

File: pkg/sdk/model/filter/grep.go [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+//    - demo-output
+// ```
+//
+// #### Fluentd Config Result
+// ```
+//  <filter **>
+//    @type grep
+//    @id demo-flow_1_grep
+//    <regexp>
+//      key elso
```
Comment (tarokkk) [inline pkg/sdk/model/filter/grep.go]:
```suggestion
//      key first
```

File: pkg/sdk/model/filter/grep.go [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+// ```
+//apiVersion: logging.banzaicloud.io/v1beta1
+//kind: Flow
+//metadata:
+//  name: demo-flow
+//spec:
+//  filters:
+//    - grep:
+//        exclude:
+//        - key: elso
```
Comment (tarokkk) [inline pkg/sdk/model/filter/grep.go]:
```suggestion
//        - key: first
```

File: pkg/sdk/model/filter/grep.go [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+//    - demo-output
+// ```
+//
+// #### Fluentd Config Result
+// ```
+//  <filter **>
+//    @type grep
+//    @id demo-flow_0_grep
+//    <exclude>
+//      key elso
```
Comment (tarokkk) [inline pkg/sdk/model/filter/grep.go]:
```suggestion
//      key first
```

File: pkg/sdk/model/filter/grep.go [hunk 6]
```diff
@@ -0,0 +1,332 @@
...
+//apiVersion: logging.banzaicloud.io/v1beta1
+//kind: Flow
+//metadata:
+//  name: demo-flow
+//spec:
+//  filters:
+//    - grep:
+//        or:
+//          - exclude:
+//            - key: elso
```
Comment (tarokkk) [inline pkg/sdk/model/filter/grep.go]:
```suggestion
//            - key: first
```

File: pkg/sdk/model/filter/grep.go [hunk 7]
```diff
@@ -0,0 +1,332 @@
...
+//              pattern: /^5\d\d$/
+//            - key: masodik
```
Comment (tarokkk) [inline pkg/sdk/model/filter/grep.go]:
```suggestion
//            - key: second
```

File: pkg/sdk/model/filter/grep.go [hunk 8]
```diff
@@ -0,0 +1,332 @@
...
+//  selectors: {}
+//  outputRefs:
+//    - demo-output
+//```
+//
+// #### Fluentd Config Result
+// ```
+//    <or>
+//      <exclude>
+//        key elso
```
Comment (tarokkk) [inline pkg/sdk/model/filter/grep.go]:
```suggestion
//        key first
```

File: pkg/sdk/model/filter/grep.go [hunk 9]
```diff
@@ -0,0 +1,332 @@
...
+//        pattern /^5\d\d$/
+//      </exclude>
+//      <exclude>
+//        key masodik
```
Comment (tarokkk) [inline pkg/sdk/model/filter/grep.go]:
```suggestion
//        key second
```

File: pkg/sdk/model/filter/grep.go [hunk 10]
```diff
@@ -0,0 +1,332 @@
...
+//apiVersion: logging.banzaicloud.io/v1beta1
+//kind: Flow
+//metadata:
+//  name: demo-flow
+//spec:
+//  filters:
+//    - grep:
+//        and:
+//          - regexp:
+//            - key: elso
```
Comment (tarokkk) [inline pkg/sdk/model/filter/grep.go]:
```suggestion
//            - key: first
```

File: pkg/sdk/model/filter/grep.go [hunk 11]
```diff
@@ -0,0 +1,332 @@
...
+//              pattern: /^5\d\d$/
+//            - key: masodik
```
Comment (tarokkk) [inline pkg/sdk/model/filter/grep.go]:
```suggestion
//            - key: second
```

File: pkg/sdk/model/filter/grep.go [hunk 12]
```diff
@@ -0,0 +1,332 @@
...
+//  selectors: {}
+//  outputRefs:
+//    - demo-output
+//```
+//
+// #### Fluentd Config Result
+// ```
+//    <and>
+//      <regexp>
+//        key elso
```
Comment (tarokkk) [inline pkg/sdk/model/filter/grep.go]:
```suggestion
//        key first
```

File: pkg/sdk/model/filter/grep.go [hunk 13]
```diff
@@ -0,0 +1,332 @@
...
+//        pattern /^5\d\d$/
+//      </regexp>
+//      <regexp>
+//        key masodik
```
Comment (tarokkk) [inline pkg/sdk/model/filter/grep.go]:
```suggestion
//        key second
```

**Pull request merged**